sdk/communication/communication-short-codes/src/shortCodesClient.ts

```diff
@@ -0,0 +1,192 @@
+import {
+  ShortCode,
+  ShortCodesGetShortCodesOptionalParams,
+  ShortCodesUpsertUSProgramBriefOptionalParams,
+  USProgramBrief
+} from "./generated/src/models/";
+import { SpanStatusCode } from "@azure/core-tracing";
+import {
+  DeleteUSProgramBriefOptions,
+  GetUSProgramBriefOptions,
+  SubmitUSProgramBriefOptions
+} from "./models";
+
+/**
+ * Client options used to configure the ShortCodesClient API requests.
+ */
+export interface ShortCodesClientOptions extends PipelineOptions {}
+
+const isShortCodesClientOptions = (options: any): options is ShortCodesClientOptions =>
+  options && !isKeyCredential(options) && !isTokenCredential(options);
+
+export class ShortCodesClient {
+  /**
+   * A reference to the auto-generated ShortCodes HTTP client.
+   */
+  private readonly client: GeneratedClient;
+
+  public constructor(connectionString: string, options?: ShortCodesClientOptions);
+
+  public constructor(url: string, credential: KeyCredential, options?: ShortCodesClientOptions);
+
+  public constructor(url: string, credential: TokenCredential, options?: ShortCodesClientOptions);
+
+  public constructor(
+    connectionStringOrUrl: string,
+    credentialOrOptions?: KeyCredential | TokenCredential | ShortCodesClientOptions,
+    maybeOptions: ShortCodesClientOptions = {}
+  ) {
+    const { url, credential } = parseClientArguments(connectionStringOrUrl, credentialOrOptions);
+    const options = isShortCodesClientOptions(credentialOrOptions)
+      ? credentialOrOptions
+      : maybeOptions;
+    const libInfo = `azsdk-js-communication-short-codes/${SDK_VERSION}`;
+
+    if (!options.userAgentOptions) {
+      options.userAgentOptions = {};
+    }
+
+    if (options.userAgentOptions.userAgentPrefix) {
+      options.userAgentOptions.userAgentPrefix = `${options.userAgentOptions.userAgentPrefix} ${libInfo}`;
+    } else {
+      options.userAgentOptions.userAgentPrefix = libInfo;
+    }
+
+    const internalPipelineOptions: InternalPipelineOptions = {
+      ...options,
+      ...{
+        loggingOptions: {
+          logger: logger.info
+        }
+      }
+    };
+
+    const authPolicy = createCommunicationAuthPolicy(credential);
+    const pipeline = createPipelineFromOptions(internalPipelineOptions, authPolicy);
+    this.client = new ShortCodesGeneratedClient(url, pipeline).shortCodes;
+  }
+
+  public listShortCodes(
+    options: ShortCodesGetShortCodesOptionalParams = {}
+  ): PagedAsyncIterableIterator<ShortCode> {
+    const { span, updatedOptions } = createSpan("ShortCodesClient-listShortCodes", options);
+    try {
+      return this.client.listShortCodes(updatedOptions);
+    } catch (e) {
+      span.setStatus({
+        code: SpanStatusCode.ERROR,
+        message: e.message
+      });
+      throw e;
+    } finally {
+      span.end();
+    }
+  }
+
+  public async upsertUSProgramBrief(
+    programBriefId: string,
+    options: ShortCodesUpsertUSProgramBriefOptionalParams = {}
+  ): Promise<RestResponse> {
```

Comment:
We shouldn't return the RestResponse, I would expect this to be `Promise<void>`.
Brian (or Jeff?) had mentioned to migrate to Azure Core 2 which generates this differently. 

Lu had done the migration for Chat: https://github.com/Azure/azure-sdk-for-js/pull/17045